daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/RepTypeMixin.scala

```diff
@@ -0,0 +1,213 @@
+    // from the associated enum
+    val enumAttributes = restriction.enumerations.map { enum =>
+      enum.schemaDefinitionWhen(
+        enum.repValuesRaw.isEmpty && enum.repValueRangesRaw.isEmpty,
+        "All enumerations must define dfdlx:repValues and/or dfdlx:repValueRanges when dfdlx:repType (%s) is defined",
+        repType,
+      )
+
+      val repValues: Seq[DataValueNumber] =
+        enum.repValuesRaw
+          .map(repTypeElementDecl.primType.fromXMLString(_).asInstanceOf[DataValueNumber])
+      val repValueRanges: Seq[(DataValueNumber, DataValueNumber)] =
+        enum.repValueRangesRaw
+          .map(repTypeElementDecl.primType.fromXMLString(_).asInstanceOf[DataValueNumber])
+          .sliding(2, 2)
+          .map { case Seq(low, high) =>
+            enum.schemaDefinitionUnless(
+              lt.operate(low, high).getBoolean,
+              "dfdlx:repValueRanges low value (%s) must be less than high value (%s)",
+              low.value,
+              high.value,
+            )
+            (low, high)
+          }
+          .toSeq
+      val enumValue = enum.enumValueCooked.asInstanceOf[DataValueString]
+      (repValues, repValueRanges, enumValue)
+    }
+
+    // combine all rep values and ranges into a single seq of low/high tuples, sort them by
+    // their low value, and look for overlaps
+    enumAttributes
+      .flatMap { case (repValues, repRanges, _) => repValues.map(v => (v, v)) ++ repRanges }
+      .sortWith { case ((l1, _), (l2, _)) => lt.operate(l1, l2).getBoolean }
+      .reduce[(DataValueNumber, DataValueNumber)] { case ((l1, h1), (l2, h2)) =>
+        restriction.schemaDefinitionUnless(
+          lt.operate(h1, l2).getBoolean,
+          "Overlapping dfdlx:%s (%s) and dfdlx:%s (%s) found",
+          if (lt.operate(l1, h1).getBoolean) "repValueRanges" else "repValues",
+          if (lt.operate(l1, h1).getBoolean) l1.value + " " + h1.value else l1.value,
+          if (lt.operate(l2, h2).getBoolean) "repValueRanges" else "repValues",
+          if (lt.operate(l2, h2).getBoolean) l2.value + " " + h2.value else l2.value,
+        )
+        (l2, h2)
+      }
+
+    // generate the data structures for fast mapping of rep values to/from logical values
+    val parseValuesMap: Map[DataValueNumber, DataValueString] = enumAttributes
+      .filterNot(_._1.isEmpty)
+      .flatMap { case (repValues, _, enumValue) =>
+        repValues.map { repValue => (repValue, enumValue) }
+      }
+      .toMap
+    val parseRangesMap: Seq[(DataValueNumber, DataValueNumber, DataValueString)] =
+      enumAttributes
+        .filterNot(_._2.isEmpty)
+        .flatMap { case (_, repRanges, enumValue) =>
+          repRanges.map { case (low, high) => (low, high, enumValue) }
+        }
+    val unparseMap: Map[DataValueString, DataValueNumber] = enumAttributes.map {
+      case (repValues, repRanges, enumValue) =>
+        val canonicalRep = None
```

Comment:
None.orElse(x.headOption).orElse(...)

I think can be simplified to

x.headOption.orElse(...)